addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1681,6 +1681,139 @@ registry.ZoomedBackgroundShape = publicWidget.Widget.extend({
     },
 });
 
+registry.ImageShapeHoverEffet = publicWidget.Widget.extend({
+    selector: "img[data-hover-effect]",
+    disabledInEditableMode: false,
+    events: {
+        "mouseenter": "_onMouseEnter",
+        "mouseleave": "_onMouseLeave",
+    },
+
+    /**
+     * @constructor
+     */
+    init() {
+        this._super(...arguments);
+        this.lastMouseEvent = Promise.resolve();
+    },
+    /**
+     * @override
+     */
+    start() {
+        this._super(...arguments);
+        this.originalImgSrc = this.el.src;
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        if (this.originalImgSrc && (this.lastImgSrc === this.el.src)) {
+            this.el.src = this.originalImgSrc;
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onMouseEnter() {
+        if (!this.originalImgSrc) {
+            return;
+        }
+        this.lastMouseEvent = this.lastMouseEvent.then(() => new Promise((resolve) => {
+            if (!this.svgInEl) {
+                fetch(this.el.src)
+                    .then(response => response.text())
+                    .then(text => {
```

Comment:
I think we can somehow avoid that fetch: the image is loaded, I think there is something to gets its data without re-fetching.

Reply:
I haven't found how to avoid the fetch ... I asked colleagues, they don't have a solution either.

Reply:
Ok, I guess the fetch should be ok anyway, the image should be cached anyway.